src/native/shared/log_functions.cc

```diff
@@ -0,0 +1,120 @@
+#include <array>
+#include <strings.h>
+
+#include <android/log.h>
+
+#include "java-interop-logger.h"
+#include "log_types.hh"
+
+// Must match the same ordering as LogCategories
+static constexpr std::array<const char*, 12> log_names = {
+	"*none*",
+	"monodroid",
+	"monodroid-assembly",
```

Comment:
The "duplication" between `LOG_CATEGORY_NAME_MONODROID_ASSEMBLY` and this feels odd, *especially* when `LOG_CATEGORY_NAME_MONODROID_ASSEMBLY` isn't used.

Reply:
Seeing as this PR was getting bigger and bigger, I didn't implement the changes I want to  make to all the `log*` functions in this file. Treat this as "infrastructure preparation" as well.